### ISSUE TITLE
0,5 Gb database size after xAnalyzer

### ISSUE THREAD

**Issue by u07** [ThunderCls/xAnalyzer#67]
The executable is 150 Mb, I try to view the winapi call arguments, so I press "analyse **one function**" or **the selected piece of code**. 

Result: 
- 4053008 xrefs found, 
- 84179 called functions populated, 
- Total Functions Detected: **1**,
- Total Comments Set: **5**. 

The analysis takes 100 seconds, the database grows up to 0,5 Gb (uncompressed json), loading it takes x64dbg 15 seconds (and the next process restart will make me wait about 30 seconds). 

All I wanted is to populate params (5 pcs). I believe... Some optimizations are strongly required here :-)

The database look like this:

![image](https://github.com/user-attachments/assets/73ec29a6-231d-4c5c-9552-82daa028e941)

**Comment by ThunderCls** [comment]
Yes, the bigger the executable, the more comments and annotations xAnalyzer will add making the database bigger. There is no "optimization" per say that can fix this, this is how the plugin works, by adding metadata to the executable (comments, functions definitions, etc). As a workaround, instead of making a full analysis on the executable, disable the automatic analysis and use the "Analyze Selection/Function" options manually when needed for a more localized analysis, which makes the plugin add less metadata and therefore a smaller database.

**Comment by u07** [comment]
@ThunderCls , please, pay attention to the part in bold: I do analyze section/function. 

That's the problem.

**Comment by ThunderCls** [comment]
@u07 the amount of xrefs and called functions flagged by xdbg is really high for just a few lines of analysis.
